modus-gh-issue-summarizer/main.go

```diff
@@ -0,0 +1,391 @@
+			"Content-Type":  "application/json",
+		},
+		Body: payload,
+	}
+
+	request := http.NewRequest("https://api.github.com/graphql", options)
+	response, err := http.Fetch(request)
+	if err != nil {
+		return "", fmt.Errorf("error fetching discussion category ID: %w", err)
+	}
+
+	var result struct {
+		Data struct {
+			Node struct {
+				DiscussionCategories struct {
+					Nodes []struct {
+						ID string `json:"id"`
+					} `json:"nodes"`
+				} `json:"discussionCategories"`
+			} `json:"node"`
+		} `json:"data"`
+	}
+
+	if err := json.Unmarshal(response.Body, &result); err != nil {
+		return "", fmt.Errorf("error parsing discussion category ID response: %w", err)
+	}
+
+	// Return the first available category
+	if len(result.Data.Node.DiscussionCategories.Nodes) > 0 {
+		return result.Data.Node.DiscussionCategories.Nodes[0].ID, nil
+	}
+
+	return "", fmt.Errorf("no discussion categories found in repository")
+}
+
+func generateKBArticle(issue *GitHubIssue, comments []GitHubComment) (string, error) {
+	prompt := fmt.Sprintf(`
+		Generate a detailed markdown article givin a concise summary of the following GitHub issue. Include the problem, the solution (if on exists), and any other relevant details. Please mention the users involved and any significant involvement in the issue.
+		
+		### Issue Details:
+		- **Title**: %s
+		- **Description**: %s
+		- **State**: %s
+		- **Created by**: %s
+		- **Created at**: %s
+		- **Labels**: %v
+		- **Reactions**: 👍 (%d), 👎 (%d), ❤️ (%d), 🎉 (%d), 🚀 (%d), 👀 (%d)
+
+		### Comments:
+		%s
+
+		Generate the output in markdown format.
+	`,
+		issue.Title, issue.Body, issue.State, issue.User.Login, issue.CreatedAt,
+		getLabelNames(issue.Labels),
+		issue.Reactions.PlusOne, issue.Reactions.MinusOne, issue.Reactions.Heart,
+		issue.Reactions.Hooray, issue.Reactions.Rocket, issue.Reactions.Eyes,
+		formatComments(comments),
+	)
+
+	model, err := models.GetModel[openai.ChatModel]("generate-article")
+	if err != nil {
+		return "", fmt.Errorf("error fetching model: %w", err)
+	}
+
+	input, err := model.CreateInput(
+		openai.NewSystemMessage("You are an assistant that generates markdown documentation."),
+		openai.NewUserMessage(prompt),
+	)
+	if err != nil {
+		return "", fmt.Errorf("error creating model input: %w", err)
+	}
+
+	input.Temperature = 0.7
+
+	output, err := model.Invoke(input)
+	if err != nil {
+		return "", fmt.Errorf("error invoking model: %w", err)
+	}
+
+	return strings.TrimSpace(output.Choices[0].Message.Content), nil
+}
+
+func getLabelNames(labels []GitHubLabel) string {
+	names := []string{}
+	for _, label := range labels {
+		names = append(names, label.Name)
+	}
+	return strings.Join(names, ", ")
+}
+
+func formatComments(comments []GitHubComment) string {
+	formatted := []string{}
+	for _, comment := range comments {
+		formatted = append(formatted, fmt.Sprintf("- **%s** (%s): %s", comment.User.Login, comment.CreatedAt, comment.Body))
+	}
+	return strings.Join(formatted, "\n")
```

Comment:
Use a string builder here:
```suggestion
	buf := strings.Builder{}
	for _, comment := range comments {
		buf.WriteString(fmt.Sprintf("- **%s** (%s): %s\n", comment.User.Login, comment.CreatedAt, comment.Body))
	}
	return buf.String()
```